testsuite/tests/lib-runtime-events/test_create_cursor_failures.ml

```diff
@@ -0,0 +1,57 @@
+(* TEST
+ include unix;
+ include runtime_events;
+ hasunix;
+ {
+   bytecode;
+ }{
+   native;
+ }
+*)
+
+(* Tests that [create_cursor]:
+ * - fails on [None] if runtime events haven't been started
+ * - doesn't double-free when it fails to attach to [None]
+ * - does manage to attach to this process if we provide the right PID
+ *)
+
+let create_and_free ?(pid) () =
+  try
+    let dir_and_pid = Option.map (fun p -> ".", p) pid in
+    let cur = Runtime_events.create_cursor dir_and_pid in
+    Runtime_events.free_cursor cur;
+    print_endline "OK"
+  with Failure msg -> print_endline msg
+
+let start_and_pause () =
+  Runtime_events.start ();
+  Runtime_events.pause ()
+
+(* workaround for finding the events file even on Windows, where
+   [Unix.getpid] doesn't match the one used to open the file *)
+let find_events_file =
+  let list_events_files () =
+    List.filter
+      (String.ends_with ~suffix:".events")
+      (Array.to_list @@ Sys.readdir ".")
+  in
+  let startup_events_files = list_events_files () in
+  fun () ->
+  let is_new_event_file f = not (List.mem f startup_events_files) in
+  List.find is_new_event_file (list_events_files ())
```

Comment:
The code here looks fairly optimistic in terms of the expected semantics of the filesystem. It assumes that all `.events` file from past test runs are visible in the filesystem when this program starts (but, as far as I know, there is no `fsync` call anywhere in the runtime that would give this guarantee), and then it assumes that the `.events` file created by the current process at runtime-events startup time is already visible in the filesystem. (The latter is not the cause of the weird test failure in this PR I guess, otherwise we would see a `Not_found` exception.)

Reply:
Isn't there a guarantee that files created by a process are subsequently visible to that process? Huh.

Reply:
Hm, I must misremember those things and your interpretation is plausible. But that would only cover the second issue I mentioned, not the accuracy of the first call to `list_event_files` that happens at program startup and lists "startup files" from other processes, right?

Reply:
I quite agree: `Runtime_events` should have a function to return the path of the current process's events file (possibly a function on a `cursor`, returning the file name, as one can readily make a cursor for the current process).

Reply:
On reflection, there's no reason for it to be a function on a cursor.

Reply:
OK, I've tidied that up and added `Runtime_events.path` to address your points about finding the events file.